### PR TITLE
Prettydiff tsx

### PR DESCRIPTION
**Pull request by szeck87**
### What does this implement/fix? Explain your changes.
Allows prettydiff to beautify tsx files and makes it the default to do so
...

### Does this close any currently open issues?

...

### Any other comments?
Replaces typescript-formatter as default for tsx
...

### Checklist

Check all those that are applicable and complete.

- [X] Merged with latest `master` branch
- [X] Regenerate documentation with `npm run docs`
- [ ] Add change details to `CHANGELOG.md` under "Next" section
- [X] Added examples for testing to [examples/ directory](examples/)
- [ ] Travis CI passes (Mac support)
- [X] AppVeyor passes (Windows support)

### CODE REVIEW

File: src/languages/tsx.coffee [hunk 1]
```diff
@@ -18,4 +18,6 @@ module.exports = {
     "tsx"
   ]
 
+  defaultBeautifier: "Pretty Diff"
```
Comment (Glavin001) [inline src/languages/tsx.coffee]: I am always concerned with switching defaults.

**Comment by szeck87** [comment]
Linux builds look fine, I'm testing removing homebrew/science tap for mac to see if it still breaks.

File: examples/simple-jsbeautifyrc/tsx/expected/test.tsx [hunk 2]
```diff
@@ -1,12 +1,10 @@
 class Test extends React.Component<Foo> {
   render() {
-    return (
-      <div className= "class" >
-      <h2 className="anotherClass" >
-        { this.foo.bar }
-        < /h2>
```
Comment (Glavin001) [inline examples/simple-jsbeautifyrc/tsx/expected/test.tsx]: This is strange whitespace.

File: examples/simple-jsbeautifyrc/tsx/expected/test.tsx [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
-    { this.foo.bar.children }
-    </div>
-    );
+    return (<div className="class">
```
Comment (Glavin001) [inline examples/simple-jsbeautifyrc/tsx/expected/test.tsx]: Concerned about switching from newline to no-newline here.
Is there a configuration option for this?

File: examples/simple-jsbeautifyrc/tsx/expected/test.tsx (hunk 2 above)
Comment (szeck87) [reply examples/simple-jsbeautifyrc/tsx/expected/test.tsx]: The `< /h2>`?  That was removed.

**Comment by Glavin001** [comment]
Here are the supported options for TypeScript Formatter: https://github.com/vvakame/typescript-formatter#read-settings-from-files

Currently, the supported options in Atom-Beautify are "all" (via `true`): https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/typescript-formatter.coffee#L8-L9

This should be changed.

File: src/languages/tsx.coffee (hunk 1 above)
Comment (szeck87) [reply src/languages/tsx.coffee]: tsx is new and not user facing yet, so it shouldn't be a breaking change.

**Comment by szeck87** [comment]
I'm going to spend a couple hours fixing typescript-formatter issues instead.  prettydiff looks to only support it on 2.2.9, and npm is on 2.2.7.

**Comment by Glavin001** [comment]
@prettydiff could Pretty Diff 2.2.9 be published to `npm`?

**Comment by szeck87** [comment]
The options being used in typescript-formatter are actually here:  https://github.com/vvakame/typescript-formatter/blob/master/lib/utils.ts#L8-L33.

The problem looks to be with the [first arg](https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/typescript-formatter.coffee#L30) we pass to typescript-formatter being `''`.  From what I can tell it interprets the file as regular TypeScript and not TSX.

**Comment by szeck87** [comment]
I pushed a fix to a new branch and looks much better:  https://github.com/szeck87/atom-beautify/commit/6c38d3df3f8005ec64434a229bec6f5245f1910c

**Pull request closed without merging**